modules/programs/rbw.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.programs.rbw;
+
+  jsonFormat = pkgs.formats.json { };
+
+  settingsModule = with lib;
+    types.submodule {
+      options = {
+        email = mkOption {
+          type = types.str;
+          example = "name@example.com";
+          description = "The email address for your bitwarden account.";
+        };
+
+        base_url = mkOption {
+          type = types.str;
+          default = null;
+          example = "bitwarden.example.com";
+          description =
+            "The base-url for a self-hosted bitwarden installation.";
+        };
+
+        identity_url = mkOption {
+          type = types.str;
+          default = null;
+          example = "identity.example.com";
+          description = "The identity url for your bitwarden installation.";
+        };
+
+        lock_timeout = mkOption {
+          type = types.ints.unsigned;
+          default = 3600;
+          example = 300;
+          description = ''
+            The amount of time that your login information should be cached.
+          '';
+        };
+
+        pinentry = mkOption {
+          type = types.package;
+          default = pkgs.pinentry;
+          defaultText = literalExample "pkgs.pinentry";
+          example = literalExample "pkgs.pinentry-gtk2";
+          description = ''
+            The pinentry program to use for password prompts.
+          '';
+        };
```

Comment:
I wonder if it's worth unifying the pinentry config at some point. There's a lot of applications that rely on it (or may rely on it), so it doesn't make much sense that it's really only in the GPG agent config right now.

Reply:
Sure, I'll copy-paste it

Reply:
> I wonder if it's worth unifying the pinentry config at some point. There's a lot of applications that rely on it (or may rely on it), so it doesn't make much sense that it's really only in the GPG agent config right now.

I was thinking the same thing but I don't know where it would fit.